File: app/src/main/java/com/roombooking/ui/home/adapter/RoomsListAdapter.java
package com.roombooking.ui.home.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.roombooking.R;
import com.roombooking.model.Item;
import com.roombooking.model.ItemParcelable;
import com.roombooking.model.SendPassModel;
import com.roombooking.repository.RetrofitClient;
import com.roombooking.ui.roomdetail.RoomDetailActivity;
import com.squareup.picasso.Picasso;

import java.util.List;


public class RoomsListAdapter extends RecyclerView.Adapter<RoomsListAdapter.RoomListViewHolder> {

    private List<Item> itemList;
    private Context context;
    private String dateStr;

    public RoomsListAdapter(Context context, List<Item> itemList) {
        this.itemList = itemList;
        this.context = context;

    }

    public void notifyData(List<Item> itemList){
        this.itemList = itemList;
        notifyDataSetChanged();
    }

    @Override
    public RoomListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.room_view_row, null);
        RoomListViewHolder roomListViewHolder = new RoomListViewHolder(layoutView);
        return roomListViewHolder;
    }

    @Override
    public void onBindViewHolder(RoomListViewHolder holder, final int position) {
        Item roomInfo = getItem(position);
        holder.capacity.setText(context.getText(R.string.capacity)+" " +roomInfo.getCapacity());
        String floor = roomInfo.getLocation();

        holder.floor.setText(floor);
        holder.roomName.setText("Room "+roomInfo.getName());
        if(roomInfo.getImages()!=null) {
            Picasso.with(context).load(RetrofitClient.API_BASE_URL+roomInfo.getImages().get(0)).into(holder.thumbnailIv);
        }

      holder.itemView.setOnClickListener(new View.OnClickListener() {
          @Override
          public void onClick(View v) {
              Item items = getItem(position);
              ItemParcelable itemParcel= new ItemParcelable();
              itemParcel.setName(items.getName());
              itemParcel.setLocation(items.getLocation());
              itemParcel.setEquipment(items.getEquipment());
              itemParcel.setImages(items.getImages());
              itemParcel.setAvail(items.getAvail());
              itemParcel.setCapacity(items.getCapacity());
              itemParcel.setSize(items.getSize());


              Intent intent = new Intent(context, RoomDetailActivity.class);
              intent.putExtra(RoomDetailActivity.PARCEL,itemParcel);
              intent.putExtra(RoomDetailActivity.DATE_STRING,dateStr);
              context.startActivity(intent);
          }
      });

    }



    @Override
    public int getItemCount() {
        return this.itemList.size();
    }

    private Item getItem(int position) {
        return itemList.get(position);
    }

    public void setDateStr(String dateStr) {
        this.dateStr = dateStr;
    }


    class RoomListViewHolder extends RecyclerView.ViewHolder {

        public ImageView thumbnailIv;
        public TextView capacity;
        public TextView roomName;
        public TextView floor;
        public View itemView;



        public RoomListViewHolder(View itemView) {
            super(itemView);
            this.itemView = itemView;
            thumbnailIv = (ImageView) itemView.findViewById(R.id.room_photo);
            capacity = (TextView) itemView.findViewById(R.id.capacity);
            roomName = (TextView) itemView.findViewById(R.id.roomname);
            floor = (TextView) itemView.findViewById(R.id.floor);


        }



    }
}

File: app/src/main/java/com/roombooking/ui/home/adapter/RoomsListAdapter.kt
package com.roombooking.ui.home.adapter

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.roombooking.R
import com.roombooking.model.Item
import com.roombooking.model.ItemParcelable
import com.roombooking.repository.RetrofitClient
import com.roombooking.ui.roomdetail.RoomDetailInfoActivity
import com.squareup.picasso.Picasso

/**
 * Created by raghavendra.malgi on 28-11-2017.
 */
class RoomsListAdapter(var context: Context, var itemList: List<Item>?) :
        RecyclerView.Adapter<RoomsListAdapter.RoomListViewHolder>() {


    private var dateStr: String? = null

    fun notifyData(itemList: List<Item>) {
        this.itemList = itemList
        notifyDataSetChanged()
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RoomsListAdapter.RoomListViewHolder {

        val layoutView = LayoutInflater.from(parent.context).inflate(R.layout.room_view_row, null)
        return RoomListViewHolder(layoutView)
    }


    override fun onBindViewHolder(holder: RoomListViewHolder, position: Int) {
        val roomInfo = getItem(position)
        holder.capacity.text = context.getText(R.string.capacity).toString() + " " + roomInfo.capacity
        val floor = roomInfo.location

        holder.floor.text = floor
        holder.roomName.text = "Room " + roomInfo.name
        if (roomInfo.images != null) {
            Picasso.with(context).load(RetrofitClient.API_BASE_URL + roomInfo.images[0]).into(holder.thumbnailIv)
        }

        holder.itemView.setOnClickListener({
            val items = getItem(position)
            val itemParcel = ItemParcelable()
            itemParcel.name = items.name
            itemParcel.location = items.location
            itemParcel.equipment = items.equipment
            itemParcel.images = items.images
            itemParcel.avail = items.avail
            itemParcel.capacity = items.capacity
            itemParcel.size = items.size


            val intent = Intent(context, RoomDetailInfoActivity::class.java)
            intent.putExtra(RoomDetailInfoActivity.PARCEL, itemParcel)
            intent.putExtra(RoomDetailInfoActivity.DATE_STRING, dateStr)
            context.startActivity(intent)
        })

    }


    override fun getItemCount(): Int {
        return this.itemList!!.size
    }

    private fun getItem(position: Int): Item {
        return itemList!![position]
    }

    fun setDateStr(dateStr: String) {
        this.dateStr = dateStr
    }

    inner class RoomListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var thumbnailIv: ImageView = itemView.findViewById(R.id.room_photo) as ImageView
        var capacity: TextView = itemView.findViewById(R.id.capacity) as TextView
        var roomName: TextView = itemView.findViewById(R.id.roomname) as TextView
        var floor: TextView = itemView.findViewById(R.id.floor) as TextView


    }

}